### PR TITLE
Add model tests

### PR DESCRIPTION
**Pull request by brannondorsey**
- Add `RW_NO_SERVE` environment variable and `no_serve` keyword argument to `runway.run()`. These settings prevent `runway.run()` from starting the Flask server so that mock HTTP requests can be made via `app.test_client()`. See [Testing Flask Applications](http://flask.pocoo.org/docs/1.0/testing/) for more details.
- Add model tests in [`tests/test_model.py`](tests/test_model.py)
- Minor change to `docs/` so that JavaScript HTTP -> HTTPS redirect only occurs when the protocol is actually `http:`.
- Update / start CHANGELOG 😜

### CODE REVIEW

**Comment by brannondorsey** [comment]
Forgot tests are now running for Python 2.7 too. Will update 🕐...

File: tests/test_model.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    meta_before = os.environ.get('RW_META')
+    os.environ['RW_META'] = '1'
+
+    rw.run(debug=True, model_options={ 'initialization_array': ['one', 'two', 'three'] })
+    std = capsys.readouterr()
+    assert json.loads(std.out) == expected_manifest
+    assert std.err == ''
+
+    if meta_before is not None: os.environ['RW_META'] = meta_before
+
```
Comment (agermanidis) [inline tests/test_model.py]: Might be worth adding two more tests for detecting `SetupError` and `InferenceError`.

Comment (brannondorsey) [reply tests/test_model.py]: 100%! There are quite a few more model tests to add actually, but I figured some is better than none. I'll add these as well.
